### PR TITLE
OPS-12937 - adding optional caseId and plan recalc

### PR DESCRIPTION
**Pull request by DreamcastKnight**

### CODE REVIEW

File: test/controllers/PegaControllerSpec.scala [hunk 1]
```diff
@@ -237,7 +237,7 @@ class PegaControllerSpec extends ItSpec with TdBase {
             PegaStub.stubOauthToken(Right(context.tdAll.pegaOauthToken))
             PegaStub.stubStartCase(Right(context.tdAll.pegaStartCaseResponse))
 
-            val result = controller.startCase(context.tdAll.journeyId)(context.request)
+            val result = controller.startCase(context.tdAll.journeyId, recalculationNeeded = true)(context.request)
```
Comment (AndyHWChung) [inline test/controllers/PegaControllerSpec.scala:240]: it'd be good to have a test around this area to see if we're passing down the new fields correctly. The start case request JSON is tested with this below:
```scala
PegaStub.verifyStartCaseCalled(
              context.tdAll.pegaOauthToken,
              expectedRequestJson
            )
```

File: app/journey/UpdatePegaStartCaseResponseController.scala [hunk 2]
```diff
@@ -60,21 +60,25 @@ class UpdatePegaStartCaseResponseController @Inject() (
         j.into[Epaye.StartedPegaCase]
           .withFieldConst(_.stage, Stage.AfterStartedPegaCase.StartedPegaCase)
           .withFieldConst(_.startCaseResponse, startCaseResponse)
+          .withFieldConst(_.pegaCaseId, Some(startCaseResponse.caseId))
```
Comment (AndyHWChung) [inline app/journey/UpdatePegaStartCaseResponseController.scala:63]: it'd be nice to have the tests for this controller test that the pegaCaseId is populated correctly in the updated journey. There should already be some tests you can add to I think

Comment (DreamcastKnight) [reply app/journey/UpdatePegaStartCaseResponseController.scala:63]: there are tests that check the whole journey that also means checking the pegacaseId, no? been looking through them

Comment (AndyHWChung) [reply app/journey/UpdatePegaStartCaseResponseController.scala:63]: it's in this controller specifically that the case id is being populated in the journey - so there should be some cases in that controller spec where you start off with a journey without a case id in it and then you should see it appear after doing the update

File: test/controllers/PegaControllerSpec.scala [hunk 3]
```diff
@@ -278,6 +320,15 @@ class PegaControllerSpec extends ItSpec with TdBase {
             )
           }
 
+          "EPAYE after started pegacase" in new JourneyItTest {
```
Comment (AndyHWChung) [inline test/controllers/PegaControllerSpec.scala:323]: the relcalculation flag isn't controlled by the journey stage - it's controlled by what is passed into the controller call in `testSuccess`. What might make more sense is to not have a new `expectedEpayeStartCaseRequestAfterStartedPegaCaseJson` and just pass in the recalc flag to `testSuccess` and also to the existing `expectedEpayeStartCaseRequestJson` to make sure the right value is passed through

Comment (DreamcastKnight) [reply test/controllers/PegaControllerSpec.scala:323]: but whether or not the caseId/the recalc needed are there in the first place is dependent on when in the journey we are, right? I can still do this to make sure the right value is passed, though

Comment (AndyHWChung) [reply test/controllers/PegaControllerSpec.scala:323]: the controller doesn't really care where you are in the journey - it just acts upon whatever info you feed it. So the BE could still receive a start case response with the flag equal to true or false if the journey is after started PEGA case - and it does depending on if you've changed an answer to the "why can't you pay in full" page or something else. But the main point is the BE doesn't know about that - that is controlled by the FE. What we're testing here is the BE can handle whatever the FE gives it

Comment (DreamcastKnight) [reply test/controllers/PegaControllerSpec.scala:323]: unless I am misunderstanding something

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**